app/src/androidTest/java/ServiceTests/PingService.java

```diff
@@ -0,0 +1,95 @@
+
+        // Set config
+        locationConfigs =
+                new LocationConfigs(TestConstants.MIN_TIME_INTERVAL_BETWEEN_TWO_LOCATIONS_PS
+                        , TestConstants.MIN_DISTANCE_INTERVAL_BETWEEN_TWO_LOCATIONS_PS
+                        , TestConstants.MIN_PING_SERVICE_SYNC_INTERVAL_PS, TestConstants.ACCURACY_PS
+                        , TestConstants.BUFFER_SIZE_PS, TestConstants.BATCH_SIZE_FOR_PING_SERVICE_PS
+                        , TestConstants.SERVICE_TIMEOUT_GLOBAL, TestConstants.XAPI_KEY_GLOBAL, TestConstants.GPS_PIPELINE_URL
+                        , TestConstants.NOTIFICATION_ICON_ID);
+
+
+        // Initiate Both Location Services
+        ServiceHelper.startBothServicesIfNotRunning(activityTestRule.getActivity().getApplication()
+                , locationConfigs, locationPingServiceCallback, appContext);
+
+        mainApplication = activityTestRule.getActivity().getApplication();
+
+    }
+
+
+    @AutoTest_PingLocationService
+    @Test
+    public void verifyFailedResponseOfPingService() {
+
+        // --------------------- Set and Validate First Location ---------------------
+        loc1 = new Location(UiUtils.randomGenerator(TestConstants.minValue, TestConstants.maxValue)
+                , UiUtils.randomGenerator(TestConstants.minValue, TestConstants.maxValue), 3);
+        AssertUtils.assertTrueV(
+                DBHelper.setLocationAndValidateDB(loc1, mainApplication),
+                "Database state does not match with the desired state",
+                "Successfully validated expected database state ");
+
+
+        // --------------------- Set and Validate Second Location ---------------------
+        loc2 = new Location(UiUtils.randomGenerator(TestConstants.minValue, TestConstants.maxValue)
+                , UiUtils.randomGenerator(TestConstants.minValue, TestConstants.maxValue), 3);
+        AssertUtils.assertTrueV(
+                DBHelper.setLocationAndValidateDB(loc2, mainApplication),
+                "Database state does not match with the desired state",
+                "Successfully validated expected database state ");
+
+        edgeCaseResponses.put("/" + TestConstants.GPS_PIPELINE_URL_END_POINT, TestConstants.RESPONSE_TYPE.SUCCESS);
+        UiUtils.safeSleep(5);
+        mockLocationList.remove(0);
+        AssertUtils.assertTrueV(DBHelper.validateLocationsDataInDatabase(mockLocationList , mainApplication),
+                "Database state does not match with the desired state",
+                "Successfully validated expected database state ");
+
+
+        UiUtils.safeSleep(6);
+        gpsLocationsCurrentlyInDatabase = DBHelper.fetchGpsDataFromDatabase(mainApplication);
+        AssertUtils.assertTrueV(gpsLocationsCurrentlyInDatabase.isEmpty(),
+                "Database state does not match with the desired state",
+                "Successfully validated expected database state ");
+
```

Comment:
Think of a failure condition for this test. Then print test pass/fail logs based upon the state of that.

Reply:
Done for all tests .